cmake/helpers.cmake

```diff
@@ -53,22 +53,19 @@ endfunction()
 function(add_umf_target_link_options name)
     if(NOT MSVC)
         if (NOT APPLE)
-            target_link_options(${name} PRIVATE "LINKER:-z,relro,-z,now")
+            set_target_properties(${name} PROPERTIES 
+                LINK_FLAGS "-z,relro,-z,now")
```

Comment:
There are many warnings, because of this line:
```
[ 62%] Linking CXX executable umf_test-c_api_disjoint_pool
/usr/bin/ld: warning: -z ,relro,-z,now ignored
[ 62%] Built target umf_test-c_api_disjoint_pool
[ 64%] Building CXX object test/CMakeFiles/umf_test-base.dir/base.cpp.o
[ 67%] Linking CXX executable umf_test-base
/usr/bin/ld: warning: -z ,relro,-z,now ignored
[ 67%] Built target umf_test-base
[ 70%] Building CXX object test/CMakeFiles/umf_test-memoryPool.dir/memoryPoolAPI.cpp.o
[ 72%] Linking CXX executable umf_test-memoryPool
/usr/bin/ld: warning: -z ,relro,-z,now ignored
[ 72%] Built target umf_test-memoryPool
[ 75%] Building CXX object test/CMakeFiles/umf_test-memoryProvider.dir/memoryProviderAPI.cpp.o
[ 78%] Linking CXX executable umf_test-memoryProvider
/usr/bin/ld: warning: -z ,relro,-z,now ignored
[ 78%] Built target umf_test-memoryProvider
[ 81%] Building CXX object test/CMakeFiles/umf_test-disjointPool.dir/disjoint_pool.cpp.o
[ 83%] Linking CXX executable umf_test-disjointPool
/usr/bin/ld: warning: -z ,relro,-z,now ignored
[ 83%] Built target umf_test-disjointPool
[ 86%] Building CXX object test/CMakeFiles/umf_test-disjointCoarseMallocPool.dir/disjointCoarseMallocPool.cpp.o
[ 89%] Linking CXX executable umf_test-disjointCoarseMallocPool
/usr/bin/ld: warning: -z ,relro,-z,now ignored
```

Reply:
fixed